cognite_toolkit/_cdf_tk/utils/producer_worker.py

```diff
@@ -22,6 +25,10 @@
 T_Download = TypeVar("T_Download", bound=Sized)
 T_Processed = TypeVar("T_Processed", bound=Sized)
 
+# Sentinels for signaling finish
+PROCESS_FINISH_SENTINEL = object()
+WRITE_FINISH_SENTINEL = object()
```

Comment:
I am replacing the non-tread safe flags with these SENTINEL to shut down the process and write workers.